Chess/common_types.h

```diff
@@ -53,10 +53,17 @@ struct CellInfo {
 
 class ChessEx: public std::exception
 {
-   const char* what() {
+   virtual const char* what() {
        return "Chess exception occurred";
```

Comment:
Not sure it's a good idea to return local temporary variable as a char* from function? Where does this pointer points after function finishes its execution?
It still works, because string literals mentioned in code are indeed stored in some fixed memory area, but still I would consider it to be a bed idea.